src/Microsoft.VisualBasic/tests/VB/DriveInfo.Windows.Tests.vb

```diff
@@ -0,0 +1,174 @@
+        Private Shared Sub DoDriveCheck()
+            ' Get Volume Label - valid drive
+            Dim serialNumber, maxFileNameLen, fileSystemFlags As Integer
+            Dim volNameLen As Integer = 50
+            Dim fileNameLen As Integer = 50
+            Dim volumeName As StringBuilder = New StringBuilder(volNameLen)
+            Dim fileSystemName As StringBuilder = New StringBuilder(fileNameLen)
+
+            Dim validDrive As IO.DriveInfo = FileSystem.Drives().First(Function(d) d.DriveType = IO.DriveType.Fixed)
+            Dim volumeInformationSuccess As Boolean = GetVolumeInformation(validDrive.Name, volumeName, volNameLen, serialNumber, maxFileNameLen, fileSystemFlags, fileSystemName, fileNameLen)
+
+            If volumeInformationSuccess Then
+                Assert.Equal(volumeName.ToString(), validDrive.VolumeLabel)
+            Else
+                Dim name As String = validDrive.VolumeLabel
+            End If
+        End Sub
+
+        <Fact>
+        <PlatformSpecific(TestPlatforms.Windows)>
+        Public Shared Sub TestGetDrives()
+            Dim validExpectedDrives As IEnumerable(Of Char) = GetValidDriveLettersOnMachine()
+            Dim validActualDrives() As IO.DriveInfo = FileSystem.Drives().ToArray
+
+            ' Test count
+            Assert.Equal(validExpectedDrives.Count(), validActualDrives.Count())
+
+            For i As Integer = 0 To validActualDrives.Count() - 1
+                ' Test if the drive-letter is correct
+                Assert.Contains(validActualDrives(i).Name(0), validExpectedDrives)
+            Next
+        End Sub
+
+        <Fact>
+        Public Shared Sub TestDriveProperties_AppContainer()
+            Dim validDrive As IO.DriveInfo = FileSystem.Drives().Where(Function(d) d.DriveType = IO.DriveType.Fixed).First()
+            Dim isReady As Boolean = validDrive.IsReady
+            If PlatformDetection.IsInAppContainer Then
+                Assert.Throws(Of UnauthorizedAccessException)(Function() validDrive.AvailableFreeSpace)
+                Assert.Throws(Of UnauthorizedAccessException)(Function() validDrive.DriveFormat)
+                Assert.Throws(Of UnauthorizedAccessException)(Function() validDrive.TotalFreeSpace)
+                Assert.Throws(Of UnauthorizedAccessException)(Function() validDrive.TotalSize)
+                Assert.Throws(Of UnauthorizedAccessException)(Function() validDrive.VolumeLabel)
+            Else
+                Assert.NotNull(validDrive.Name)
+                Assert.NotNull(validDrive.RootDirectory.Name)
+                Assert.NotNull(validDrive.DriveFormat)
+                Assert.[True](validDrive.AvailableFreeSpace > 0)
+                Assert.[True](validDrive.TotalFreeSpace > 0)
+                Assert.[True](validDrive.TotalSize > 0)
+                Assert.NotNull(validDrive.VolumeLabel)
+            End If
+        End Sub
+
+        <Fact>
+        <PlatformSpecific(TestPlatforms.Windows)>
```

Comment:
why adding this attribute ?

Reply:
The DriveInfo test were copied from CoreFx IO Tests because VB does almost nothing except change the name of the function. I am guessing that I need to make these tests and the code OS independent if possible.

Reply:
If the VB code is just a pass thru then we only need a minimal test. No need to repeat tests, eg if it uses a dictionary we assume that has its own tests..

Reply:
@danmosemsft  Almost every test I have written is for pass through code, I am mostly testing corner cases where I have found issues like using ("C:\").

Reply:
I haven't had a look at your change yet, but in general I would only test behavior that is implemented in the VB code. If it's just checking parameters then calling through to System.IO then I would test the parameter validation plus one test to verify it calls through successfully.